cmdstanpy/stanfit.py

```diff
@@ -547,21 +560,93 @@ def get_drawset(self, params: List[str] = None) -> pd.DataFrame:
                 if not (param in self._column_names or param in pnames_base):
                     raise ValueError('unknown parameter: {}'.format(param))
         self._assemble_sample()
-        # pylint: disable=redundant-keyword-arg
-        data = self.sample.reshape(
-            (self.num_draws * self.runset.chains),
-            len(self.column_names),
-            order='A',
-        )
-        drawset = pd.DataFrame(data=data, columns=self.column_names)
+        if self._drawset is None:
+            # pylint: disable=redundant-keyword-arg
+            data = self.sample.reshape(
+                (self.num_draws * self.runset.chains),
+                len(self.column_names),
+                order='A',
+            )
+            self._drawset = pd.DataFrame(data=data, columns=self.column_names)
         if params is None:
-            return drawset
+            return self._drawset
         mask = []
         for param in params:
             for name in self.column_names:
                 if param == name or param == name.split('.')[0]:
                     mask.append(name)
```

Comment:
We can remove on for loop

```suggestion
        mask = []
        params = set(params)
        for name in self.column_names:
            if any(item in params for item in (name, name.split('.')[0]):
                mask.append(name)
```